datastream-kafka/src/main/java/com/linkedin/datastream/kafka/KafkaPassthroughRecordMagicConverter.java

```diff
@@ -0,0 +1,64 @@
+/**
+ *  Copyright 2020 LinkedIn Corporation. All rights reserved.
+ *  Licensed under the BSD 2-Clause License. See the LICENSE file in the project root for license information.
+ *  See the NOTICE file in the project root for additional information regarding copyright ownership.
+ */
+package com.linkedin.datastream.kafka;
+
+import java.io.UnsupportedEncodingException;
+
+import org.apache.kafka.common.header.internals.RecordHeader;
+import org.apache.kafka.common.header.internals.RecordHeaders;
+
+import com.linkedin.datastream.common.DatastreamRuntimeException;
+
+
+/**
+ * This class is a utility class to convert the PassthroughConsumerRecord's magic() to a String, and back into byte.
+ * It also adds a utility to create the RecordHeaders which can be used to pass headers with the magic set to the
+ * ProducerRecord for passthrough message format bump support.
+ */
+public class KafkaPassthroughRecordMagicConverter {
+  private static final String CHARSET = "ISO-8859-1";
+
+  public static final String PASS_THROUGH_MAGIC_VALUE = "__passThroughMagicValue";
+
+  /**
+   * Convert the magic byte of the PassthroughConsumerRecord to a String
+   */
+  public static String convertMagicToString(byte magic) {
+    try {
+      return new String(new byte[] { magic }, CHARSET);
+    } catch (UnsupportedEncodingException e) {
+      throw new DatastreamRuntimeException(String.format("Cannot convert magic byte %02X to String", magic), e);
+    }
+  }
+
+  /**
+   * Convert the converted magic String back into a byte array
+   */
+  public static byte[] convertMagicStringToByteArray(String magic) {
+    try {
+      return magic.getBytes(CHARSET);
```

Comment:
Can it return a single byte instead? since we are using ISO-8 we can be sure not to get multiple bytes. Makes the APIs look consistent in the encode and decode with taking 1 byte and returning 1 byte.

Reply:
oh nvm, looks like RecordHeader does require a byte array.

Reply:
yeah, that's why I decided to return a byte[] instead of byte.